### ISSUE TITLE
moment.localeData().firstDayOfWeek() is not a function

### ISSUE THREAD

**Issue by Greg-Owens** [dangrossman/bootstrap-daterangepicker#919]
Hi,
Probably just the newbie in me but I am consistently getting the error 

Uncaught TypeError: moment.locale(...).firstDayOfWeek is not a function(…)(anonymous function) @ VM3965:1InjectedScript._evaluateOn @ VM3200:875InjectedScript._evaluateAndWrap @ VM3200:808InjectedScript.evaluateOnCallFrame @ VM3200:933DateRangePicker @ daterangepicker.js:71(anonymous function) @ daterangepicker.js:1488jQuery.extend.each @ jquery.js:384jQuery.fn.jQuery.each @ jquery.js:136$.fn.daterangepicker @ daterangepicker.js:1484Template.report.rendered @ report.js:224(anonymous function) @ template.js:116Template._withTemplateInstanceFunc @ template.js:457fireCallbacks @ template.js:112(anonymous function) @ template.js:205(anonymous function) @ view.js:107Blaze._withCurrentView @ view.js:538(anonymous function) @ view.js:106Tracker._runFlush @ tracker.js:497onGlobalMessage @ setimmediate.js:102

It is occurring in line 71 of daterangepicker 

        this.locale = {
            format: 'MM/DD/YYYY',
            separator: ' - ',
            applyLabel: 'Apply',
            cancelLabel: 'Cancel',
            weekLabel: 'W',
            customRangeLabel: 'Custom Range',
            daysOfWeek: moment.weekdaysMin(),
            monthNames: moment.monthsShort(),
            firstDay: moment.localeData().firstDayOfWeek()
        };

I have 2.11.1 of moment and 2.1.13 of your daterangepicker.

Thanks in advance for your help,
Greg

**Comment by dangrossman** [comment]
Did you fail to include moment.js in your page, or did you not load it before daterangepicker.js?

**Comment by Greg-Owens** [comment]
HI,

I am developing with meteor and I have the moment package included in the packages file.  I can debug at the point in daterangepicker and the moment.localeData() is OK but there isn’t a function firstDayOfWeek.

Thanks
Greg

**Comment by dangrossman** [comment]
If you go to http://www.daterangepicker.com and type `moment.localeData().firstDayOfWeek()` in your browser console, you'll see 0, not an error.

If you go to http://momentjs.com and type `moment.localeData().firstDayOfWeek()` in your browser console, you'll see 0, not an error.

I don't think there's anything further I can do to help you. I'm not a meteor developer.

**Comment by Greg-Owens** [comment]
Thanks - will chase it down..

Greg

**Comment by zahedul** [comment]
meteor load files alphabetic order. so first load daterangepicker.js after that moment.js .
I also faced same issues.

**Comment by jamiesoncj** [comment]
Just wanted to add that I faced the same issue (again, I'm working with Meteor). Solved for me by switching from using [dangrossman:bootstrap-daterangepicker](https://atmospherejs.com/dangrossman/bootstrap-daterangepicker) package to using [gilbertwat:bootstrap3-daterangepicker](https://atmospherejs.com/gilbertwat/bootstrap3-daterangepicker). Handy hint from [this Meteor forums post](https://forums.meteor.com/t/trying-to-use-daterangepicker-in-meteor-1-2-solved/17480/6?u=jamiesoncj). Hope this helps someone else.

**Comment by chefArchitect** [comment]
@dangrossman Although this issue is purely a matter of loading sequence of modules through 'require', I think you should add a check and provide a fallback mode together with an 'incompatible version' warning or something, since many integrators of your module might have explicitly loaded moment.js in their project.

**Comment by SarasArya** [comment]
Okay one point, when developing application with angular there are two folders that get installed with bower install i.e. moment and momentjs, use the file from moment folder not the other one. That helped me to resolve the issue.

**Comment by hakanseysane** [comment]
Please check version of moment.js. Maybe it's older version.

**Comment by abelardolg** [comment]
Hi there!

I wrote `moment.localeData().firstDayOfWeek()` in the console of my browser when I am at your website and it gave an error:
![momentFails](https://user-images.githubusercontent.com/22522883/65752393-fa7f1380-e10c-11e9-8c52-a22fa69ef02d.png)

I downloaded the latest version of this library and it showed the following message:
![image](https://user-images.githubusercontent.com/22522883/65752481-31edc000-e10d-11e9-8182-c35b19953f9d.png)

...and yes, I put moment.js before your library inside my html.twig file.

Warmest regards.
